payjoin/src/send/mod.rs

```diff
@@ -1181,4 +1191,28 @@ mod test {
         let deserialized = serde_json::from_str(&serialized).unwrap();
         assert!(req_ctx == deserialized);
     }
+
+    #[test]
+    fn handle_known_errors() {
+        let ctx = create_v1_context();
+        let known_json_error = serde_json::json!({
+            "errorCode": "version-unsupported",
+            "message": "This version of payjoin is not supported."
+        })
+        .to_string();
+        assert_eq!(
+            ctx.process_response(&mut known_json_error.as_bytes()).unwrap_err().to_string(),
+            "This version of payjoin is not supported."
+        );
+        let ctx = create_v1_context();
+        let invalid_json_error = serde_json::json!({
+            "err": "random",
+            "message": "This version of payjoin is not supported."
+        })
+        .to_string();
+        assert_eq!(
+            ctx.process_response(&mut invalid_json_error.as_bytes()).unwrap_err().to_string(),
+            "The receiver sent an invalid response: couldn't decode as PSBT or JSON"
+        )
```

Comment:
I might check `Eq` between error types instead of the maginc string representation next time